src/quicklist.c

```diff
@@ -562,6 +562,7 @@ int quicklistPushHead(quicklist *quicklist, void *value, size_t sz) {
         return 1;
     }
 
+    assert(sz < UINT32_MAX); /* TODO: add support for quicklist nodes that are sds encoded (not zipped) */
```

Comment:
you should now remove these asserts in this file, and a bunch of checks in t_list.c
added them in redis 6.2 and they got cherry picked to unstable, but in your branch that's handled
please go over the change in c5e6a6204c4cf57f85e7c83a9b4e99f1a7204fd2 and undo the ones that are no longer needed.